linera-core/src/chain_worker/actor.rs

```diff
@@ -214,7 +216,25 @@ where
                     let _ = callback.send(self.worker.chain_state_view().await);
                 }
                 ChainWorkerRequest::QueryApplication { query, callback } => {
-                    let _ = callback.send(self.worker.query_application(query).await);
+                    let (execution_state_sender, execution_state_receiver) =
+                        futures::channel::mpsc::unbounded();
+                    let (request_sender, request_receiver) = std::sync::mpsc::channel();
+                    let context = self.worker.current_query_context();
+
+                    let runtime_thread = tokio::task::spawn_blocking(move || {
+                        ServiceSyncRuntime::new(execution_state_sender, context)
+                            .run(request_receiver)
+                    });
+
+                    let response = self
+                        .worker
+                        .query_application(query, execution_state_receiver, request_sender)
+                        .await;
+
+                    runtime_thread
+                        .await
+                        .expect("Service runtime thread should not panic");
+                    let _ = callback.send(response);
```

Comment:
Can we avoid `let _ =` here and elsewhere? This silents the very important warning re: unused returned values, so the reader has to figure out if every single instance is correct.

Reply:
Sure, I'll do it as a separate PR :+1: